### ISSUE TITLE
fread fails to detect numeric

### ISSUE THREAD

**Issue by MichaelChirico** [Rdatatable/data.table#2123]
Trying to read [this csv](https://raw.githubusercontent.com/fivethirtyeight/data/master/drug-use-by-age/drug-use-by-age.csv):

```
URL = paste0('https://raw.githubusercontent.com/fivethirtyeight/',
             'data/master/drug-use-by-age/drug-use-by-age.csv')
x = fread(URL, na.strings = "-")
x$`cocaine-frequency`
#  [1] "5.0"  "1.0"  "5.5"  "4.0"  "7.0"  "5.0"  "5.0"  "5.5"  "8.0"  "5.0"  "5.0"  "6.0"  "5.0"  "8.0"  "15.0" "36.0" NA  
```

Why can `fread` detect `integer` when there's an `na.string` present, but not `numeric`?

```
sapply(fread('a,b\n1,-\n2,3.1', na.strings = '-'), class)
#           a           b 
#   "integer" "character" 

sapply(fread('a,b\n1,-\n2,3', na.strings = '-'), class)
#         a         b 
# "integer" "integer" 
```

(also setting `dec = '.'` doesn't fix this)

**Comment by franknarf1** [comment]
Fwiw, on 

> data.table 1.10.5 IN DEVELOPMENT built 2017-04-04 00:37:37 UTC; travis

I see `x[['cocaine-frequency']]` as numeric when running your code.

**Comment by MichaelChirico** [comment]
Well that narrows it down, I just rebuilt before posting. @mattdowle is there something around [this](https://github.com/Rdatatable/data.table/commit/7f006b1b12050feeac2bccf8a293cc708d4df6f8) commit that I missed?

**Comment by MichaelChirico** [comment]
Closed via #2129
